lib/lib_test.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2019 TFG Co <backend@tfgco.com>
+ * Author: TFG Co <backend@tfgco.com>
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package lib_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/topfreegames/marathon/lib"
+	"gopkg.in/jarcoal/httpmock.v1"
+)
+
+var _ = Describe("Lib", func() {
+	var m lib.MarathonInterface
+
+	ctx := context.Background()
+	now := time.Now()
+	appID := uuid.NewV4()
+	template := "template"
+	jobID := uuid.NewV4()
+	config := &lib.Config{
+		Timeout:             100 * time.Millisecond,
+		MaxIdleConnsPerHost: 100,
+		MaxIdleConns:        100,
+		URL:                 "http://marathon",
+		UserEmail:           "user@email.com",
+		AppID:               appID,
+	}
+
+	BeforeSuite(func() {
+		httpmock.Activate()
+	})
+
+	BeforeEach(func() {
+		//default configs for each test
+		m = lib.NewMarathon(config)
+		httpmock.Reset()
+	})
+
+	toString := func(m interface{}) string {
+		bytes, err := json.Marshal(m)
+		if err != nil {
+			Fail(err.Error())
+		}
+
+		return string(bytes)
+	}
+
+	Describe("Creating new job", func() {
+		It("should create job", func() {
+			url := fmt.Sprintf("http://marathon/apps/%s/jobs", appID)
+			httpmock.RegisterResponder(
+				"POST", url,
+				httpmock.NewStringResponder(200, toString(map[string]interface{}{
```

Comment:
The response of a list of jobs will be `[]model.Job{}`. I think using the real response model with `httpmock.NewJsonResponder` will be better than using a `map[string]interface{}`.